bigflow/dataflow.py

```diff
@@ -0,0 +1,42 @@
+import typing
+
+from apache_beam import Pipeline
+from apache_beam.options.pipeline_options import PipelineOptions, GoogleCloudOptions
+
+from .workflow import Job, JobContext
+
+
+class BeamJob(Job):
+    def __init__(
+            self,
+            id: str,
+            driver_callable: typing.Callable[[Pipeline, JobContext, typing.Dict], None],
+            pipeline_options: PipelineOptions,
+            driver_arguments: typing.Optional[dict] = None,
+            wait_until_finish: bool = True):
+        self._id = id
+        self.driver_callable = driver_callable
+        self.driver_arguments = driver_arguments
+        self.pipeline_options = pipeline_options
+        self.wait_until_finish = wait_until_finish
+
+    def execute(self, context: JobContext):
+        pipeline_options = self._apply_logging(self.pipeline_options, context.workflow.workflow_id)
```

Comment:
JobContext.worfkflow is `Optaionl[Workflow]`
Add 'if context.workflow' or we can change type of JobContext.workflow, which might be tricky however.

Reply:
done